src/net/dmytro/homework4/Cat.java

```diff
@@ -1,31 +1,36 @@
 package net.dmytro.homework4;
 
 public class Cat extends Animal {
-private String catName;
-private static int count = 0;
+private String name;
+private static int count;
+
+    public Cat(String name) {
+        this.name=name;
+        count++;
+
 
-    public Cat(String catName) {
-        this.catName = catName;
     }
 
+
     @Override
-    void ran(int maxRan) {
-        if (maxRan >= 0 ) {
-            System.out.println("Cat " + catName +" run " + maxRan + " м");
+    void ran(int maxRanDistance, int distanceToDo) {
+        if (maxRanDistance >= distanceToDo && distanceToDo >0 ) {
+            System.out.println("Cat " + name +" run " + distanceToDo + " м");
         }
-        count++;
+
         }
 
 
     @Override
-    void swim(int maxSwim) {
-        if (maxSwim == 0) {
-            System.out.println("Cat can't swim ");
+    void swim(int maxSwimDistance, int distanceToDo) {
+        if (maxSwimDistance >= distanceToDo && distanceToDo > 0) {
```

Comment:
for cats you dont need to do any validations, it is enough just to print, that cat cannot swim. 
because in this implementation you count on developer who use this method and you think he will put 0; 
avoid such situations, if cat cannot swim, thats all, doesn't matter what developer try to do.